pkg/connection/manager.go

```diff
@@ -221,24 +221,25 @@ func (m *DetailsManager) connectStore(ctx context.Context, p *v1.PublishConnecti
 	return m.storeBuilder(ctx, m.client, sc.GetStoreConfig())
 }
 
-// SecretMustBeOwnedBy requires that the current object is a
+// SecretToWriteMustBeOwnedBy requires that the current object is a
 // connection secret that is owned by an object with the supplied UID.
-func SecretMustBeOwnedBy(so metav1.Object) store.WriteOption {
-	return func(_ context.Context, current, desired *store.Secret) error {
-		o := ""
-		if current.Metadata != nil {
-			o = current.Metadata.GetOwnerUID()
-		}
-
-		switch {
-		case o == "" && current.Metadata != nil && current.Metadata.Type != nil && desired.Metadata != nil && current.Metadata.Type != desired.Metadata.Type:
-			return errors.Errorf(errFmtRefusingUnowned, *current.Metadata.Type)
-		case o == "":
```

Comment:
Removed taking ownership of an unowned resource assuming this was for some sort of backward compatibility thing. This simplified the logic here a lot.